client/landing/gutenboarding/components/domain-categories/index.tsx

```diff
@@ -6,30 +6,44 @@ import { Button, Icon } from '@wordpress/components';
 import classNames from 'classnames';
 import { useI18n } from '@automattic/react-i18n';
 import { useState } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
-
-import { domainCategories, DomainCategory } from '../../domains-constants';
+import { DOMAIN_SUGGESTIONS_STORE } from '../../stores/domain-suggestions';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
 export interface Props {
-	onSelect: ( domainCategory?: DomainCategory ) => void;
-	selected?: DomainCategory;
+	onSelect: ( domainCategorySlug?: string ) => void;
+	selected?: string;
 }
 
 const DomainPickerCategories: React.FunctionComponent< Props > = ( { onSelect, selected } ) => {
 	const { __ } = useI18n();
+	const domainCategories = useSelect( ( select ) =>
+		select( DOMAIN_SUGGESTIONS_STORE ).getCategories()
+	);
+
+	// Sort domain categories by tier, then by title.
+	const sortedDomainCategories = [
```

Comment:
This seems like a concern for the data store, not something the component should need to worry about.

I'd prefer that we sort 1 time as the data enters the application rather than on every render.

It's the sort of thing `make-json-schema-parser` is good at.

Reply:
The parser is here:
https://github.com/Automattic/wp-calypso/blob/master/client/lib/make-json-schema-parser/index.js

Happy to help out with this.

Reply:
Thanks @sirreal! I will attempt to make a schema & transformer based on existing code I see on Calypso and will get back to you with any questions!

Reply:
Coming back to this… it's probably a big ask for this PR but something I'd like to see happen soon.

For this PR I still think any sorting should happen in the data store when the request is resolved rather than doing it in the component.

We don't need to add the complexity of using `makeJsonSchemaParser` here, which would require extraction to an isolated package, etc.

Hope that makes sense!

Reply:
Oh well for me its just a learning experience but in that case for this PR I'll just move the sorting logic upwards to the data store then wrap this up then.

Reply:
Done. 
![image](https://user-images.githubusercontent.com/1287077/82341309-8b826200-99f0-11ea-9f3f-e35a5735a0b8.png)